STYLEGUIDE.md

```diff
@@ -0,0 +1,133 @@
+
+# Typedefs styleguide
+
+This styleguide aims to answer questions relevants to the cosmetics related to 
+writing Idris programs.
+
+Since idris is a relatively young language, there is no general convention 
+about its style. This document should serve as reference for current and future 
+users of the typedefs codebase.
+
+## Line length: 120
+
+The maximum line length is 120 characters. In today's wide screen monitors 
+there is no reason to limit ourselves to 80 characters, yet, having a maximum
+limit helps avoiding degenerate situations.
+
+## Indent style: 2 spaces
+
+Idris makes uses of some whitespace sensitive construction (typically the `do`
+notation), and also inherits from a culture of ascii-art code from Agda and
+Haskell. In order to allow this level of visual customisation and ensure
+consistency across editors, spaces are used.
+
+The language does not benefit from more than 2 space of indentation in most
+cases. Thought special constructions lend themselve to more spaces for 
+indentation.
+
+### Indentation: where clauses
+
+``` idris
+fn : String
+fn = g
+  where g : String
+        g = "hello"
+```
+
+Where clauses are aligned with 6 spaces
+
+### Indentation: Let declarations
+
+```idris
+fn : String
+fn = let greeting = "hello" in
+         greeting
+```
+
+Let declarations are indented with 4 spaces
+
+### Indentation: rewrites
+
+```idris
+fn : String
+fn = rewrite proof in
+             1234
```

Comment:
Won't this lead to a "ladder" when you have successive rewrites?

Reply:
That is true. Do you suggest we have

```
fn = rewrite proof1 in
     rewrite proof2 in
             1234
```
?